spec/source_diff_spec.rb

```diff
@@ -6,18 +6,24 @@
   subject(:source_diff) { described_class.new(diff) }
   let(:diff) { Git::Diff.new(repo) }
   let(:repo) { Crystalball::GitRepo.new('.') }
-  let(:diff_file1) { Git::Diff::DiffFile.new(repo, path: 'file1.rb') }
-  let(:diff_file2) { Git::Diff::DiffFile.new(repo, path: 'file2.rb') }
+  let(:diff_file1) { instance_double('Git::Diff::DiffFile') }
+  let(:diff_file2) { instance_double('Git::Diff::DiffFile') }
+  let(:file_diff1) { double(relative_path: 'file1.rb') }
+  let(:file_diff2) { double(relative_path: 'file2.rb') }
 
   before do
+    allow(Crystalball::SourceDiff::FileDiff).to receive(:new).with(diff_file1).and_return(file_diff1)
+    allow(Crystalball::SourceDiff::FileDiff).to receive(:new).with(diff_file2).and_return(file_diff2)
+    allow(Crystalball::SourceDiff::FormattingChecker).to receive(:pure_formatting?).with(file_diff1).and_return(false)
+    allow(Crystalball::SourceDiff::FormattingChecker).to receive(:pure_formatting?).with(file_diff2).and_return(true)
     allow(diff).to receive(:each).with(no_args).and_yield(diff_file1).and_yield(diff_file2)
   end
 
   describe '#each' do
     it 'yields all changed files' do
       paths = []
       subject.each { |file_diff| paths << file_diff.relative_path }
-      expect(paths).to contain_exactly('file1.rb', 'file2.rb')
+      expect(paths).to contain_exactly('file1.rb')
     end
```

Comment:
it feels like this test does too much. Can we have a special case for formatting only?